crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -670,6 +674,7 @@ impl<'input> CanOmitOptionalParenthesesVisitor<'input> {
                 if op.is_invert() {
                     self.update_max_precedence(OperatorPrecedence::BitwiseInversion);
                 }
+                self.first.set_if_none(First::Token);
```

Comment:
Is this true for all unary operators?

Reply:
All unary operators come before the operand, so I think yes.